### ISSUE TITLE
strange bug related to serial.print/eeprom.get 

### ISSUE THREAD

**Issue by erathos** [per1234/WatchdogLog#1]
I get a strange bug in running the example sketches on my mega2560R3 board and IDE 1.6 using EEPROM.h v2:  
1) putting the line 15 "Serial.println(eepromaddress)" in comments prevents the Watchdog  reset in the WatchdogLogExample (stays into the while(true) loop at the end of setup()). Surprisingly, a Serial.println(anynumber) at line 16 restores the sensitivity to W reset.
2) Similarly, the ResetCause sketch runs fine (i.e. with W reset)  only after adding a Serial.println(anynumber) after the eeprom.get command, but stays into the while(true) without it.
3) the two native examples run the W reset independently of such Serial.println line when the five last lines of setup() are moved into the loop() section. What could be the source of such irrational bug ? Thanks

**Comment by per1234** [comment]
I'm unable to reproduce this with Arduino IDE 1.6.9, Arduino AVR Boards 1.6.11, and Mega 2560. 
- Which version of the Arduino IDE are you using?
- Which version of Arduino AVR Boards are you using? The version can be found in Tools > Board > Boards Manager.
- Have you made any modifications to the examples other than what you have described?

**Comment by erathos** [comment]
IDE 1.6.0 macosX  (the upper version running in MacOSX 10.6.8/ java 6) supplemented by  EEPROM.h v2 lib
Board: mega2560 rev3  Arduino (not a clone), bootloader = stk500v2_mega2560 from Marc Sproul 2012 (included in IDE and loaded using avr ISP mkII)
No modifications in sketches except the comment of //Serial.println(eepromaddress) which inhibits the wachdog system reset.
Thanks for your help

**Comment by per1234** [comment]
This is very strange. I can't reproduce the issue with Arduino IDE 1.6.0 either. The behavior you reported is of concern to me because it's important that the watchdog reset is not interfered with by the use of this library. Could you try the following sketch:

``` arduino
#include <WatchdogLog.h>
#include <EEPROM.h>  //If you are using an Arduino IDE version prior to 1.6.2 then you will need to install the EEPROM v2.0 library from here: https://github.com/arduino/Arduino/tree/ide-1.5.x/hardware/arduino/avr/libraries/EEPROM. If you are using Arduino IDE version 1.6.2 or greater then the library is already installed.

const unsigned int EEPROMaddress = 0;  //EEPROM address to store the program address where the watchdog timeout reset occurred. The saved program address will take 4 bytes of EEPROM starting with this address.

void setup() {
  Serial.begin(9600);
  while (!Serial) {};  //for Leonardo et al. Wait for the Serial Monitor to be opened before proceeding with the sketch.
  Serial.println("\nProgram start");
  unsigned long programAddress;
  EEPROM.get(EEPROMaddress, programAddress);
  if (programAddress > 0) {
    Serial.print("Logged program address: ");
//    Serial.println(programAddress, HEX);
    programAddress = 0;
    EEPROM.put(EEPROMaddress, programAddress);  //reset the EEPROM
  }
  else {
    Serial.println("No program address logged");
  }
  Serial.println("Waiting for watchdog timeout reset...");
  wdt_enable(WDTO_500MS);  //enable the watchdog timer with a 500 millisecond timeout duration
  WatchdogLog.begin(EEPROMaddress);  //initialize WatchdogLog
  while (true) {  //perpetual loop until the watchdog times out
    Serial.println("while loop");
  }
}

void loop() {
}
```

The idea is to determine if the while loop continues to run or if it is hanging after the watchdog times out. So let me know what you see in the Serial Monitor when you run it. The problem with this is adding the code may cause the issue to no longer occur.

**Comment by erathos** [comment]
I just resolved the problem: I have uploaded a new bootloader stk500 from other source and your sketches now run without any problem in IDE 1.6, getting a reboot inside the while loop idependently of the serial.print. This issue was so strange but  I think that  the stk500 bootloader included in arduino 1.6 is corrupted because the hex file  has some mismatches with the one I use now. Thanks anyway for your help. Your library is nice.

**Comment by per1234** [comment]
Glad to hear it's working! Strangely I still can't reproduce even after burning the bootloader included with Arduino IDE 1.6.0 to my Mega 2560. I'm going to go ahead and close this issue but please let me know if you encounter any bugs with this library. I'm glad to hear you have found it to be useful. Per
